### PR TITLE
chore: setup ESLint + run autofix

### PR DESCRIPTION
**Pull request by P0lip**
_Closes https://github.com/stoplightio/spectral/issues/1152
Needs https://github.com/stoplightio/spectral/pull/1153

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

Note - the setup is not set in the stone.
Feel free to submit your own ideas for rules.
I tried to find a right balance, so we don't have to update plenty of code right away.
We can make everything more strict as we go.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,8 +41,9 @@
     "cli:debug": "node -r ts-node/register -r tsconfig-paths/register --inspect-brk src/cli/index.ts",
     "generate-assets": "node -r ts-node/register ./scripts/generate-assets.ts",
     "inline-version": "./scripts/inline-version.js",
+    "prelint": "tsc --noEmit",
     "lint.fix": "yarn lint --fix",
-    "lint": "tsc --noEmit && tslint 'src/**/*.ts'",
```
Comment (nulltoken) [inline package.json:45]: @P0lip Don't we want to run the typechecking during the lint any more?

Comment (P0lip) [reply package.json:45]: As in the other PR - typechecking happens in `prelint` command now.

Comment (nulltoken) [reply package.json:45]: Got it!

File: src/__tests__/linter.jest.test.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ describe('Linter', () => {
     it('should have access to function-live lifespan cache', async () => {
       const logSpy = jest.spyOn(global.console, 'log').mockImplementation(Function);
 
-      await spectral.setRuleset({
+      spectral.setRuleset({
```
Comment (nulltoken) [inline src/__tests__/linter.jest.test.ts:84]: @P0lip Wow.... Caught by Eslint?

**Comment by nulltoken** [comment]
@P0lip What's the plan to fix the remaining warnings? As time goes by? Log a tech debt issue then later fail the build when warnings are detected?

**Comment by P0lip** [comment]
@nulltoken I'm going to go through them and fix most of them manually to reduce the noise.

File: src/__tests__/linter.jest.test.ts (hunk 2 above)
Comment (P0lip) [reply src/__tests__/linter.jest.test.ts:84]: Yeah!

**Comment by P0lip** [comment]
I only updated 2 lines of new code, so won't bother with you another review, @nulltoken, but letting you know to make sure you saw the changes and are okay with it.

**Pull request merged**